src/eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -112,63 +104,30 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None, input_meta=None):
-                                           meta.spec_hw[2])
-        else:
-            meta.spec_hw_range = np.array([meta.spec_hw])
-        # Increase relevant meta parameter values
-        meta.spec_hw_range *= meta.expand
-    elif hasattr(meta, 'photap'):
-        if isinstance(meta.photap, list):
-            meta.spec_hw_range = np.arange(meta.photap[0],
-                                           meta.photap[1]+meta.photap[2],
-                                           meta.photap[2])
-        else:
-            meta.spec_hw_range = np.array([meta.photap])
-        # Super sampling not supported for photometry
-        # This is here just in case someone tries to super sample
-        if meta.expand > 1:
-            print("Super sampling not supported for photometry.")
-            print("Setting meta.expand to 1.")
-            meta.expand = 1
-
-    # check for range of background apertures
-    if hasattr(meta, 'bg_hw'):
-        if isinstance(meta.bg_hw, list):
-            meta.bg_hw_range = np.arange(meta.bg_hw[0],
-                                         meta.bg_hw[1]+meta.bg_hw[2],
-                                         meta.bg_hw[2])
-        else:
-            meta.bg_hw_range = np.array([meta.bg_hw])
-        meta.bg_hw_range *= meta.expand
-    elif hasattr(meta, 'skyin') and hasattr(meta, 'skywidth'):
-        # E.g., if skyin = 90 and skywidth = 60, then the
-        # directory will use "bg90_150"
-        if not isinstance(meta.skyin, list):
-            meta.skyin = [meta.skyin]
-        else:
-            meta.skyin = range(meta.skyin[0],
-                               meta.skyin[1]+meta.skyin[2],
-                               meta.skyin[2])
-        if not isinstance(meta.skywidth, list):
-            meta.skywidth = [meta.skywidth]
-        else:
-            meta.skywidth = range(meta.skywidth[0],
-                                  meta.skywidth[1]+meta.skywidth[2],
-                                  meta.skywidth[2])
-        meta.bg_hw_range = [f'{skyin}_{skyin+skywidth}'
-                            for skyin in meta.skyin
-                            for skywidth in meta.skywidth]
+        meta = S3MetaClass(**me.mergeevents(meta, s2_meta).__dict__)
+
+    # First apply any instrument-specific defaults
+    if meta.photometry:
+        if meta.inst == 'miri':
+            meta.set_MIRI_Photometry_defaults()
+        elif meta.inst == 'nircam':
+            meta.set_NIRCam_Photometry_defaults()
+    else:
+        if meta.inst == 'miri':
+            meta.set_MIRI_defaults()
+        elif meta.inst == 'nircam':
+            meta.set_NIRCam_defaults()
+        elif meta.inst == 'nirspec':
+            meta.set_NIRSpec_defaults()
+        elif meta.inst == 'niriss':
+            meta.set_NIRISS_defaults()
+        elif meta.inst == 'wfc3':
+            meta.set_WFC3_defaults()
+    # Then apply instrument-agnostic defaults
+    meta.set_defaults()
```

Comment:
No, so I set it up so that the order of precedence is: user-set values, instrument -specific defaults, and generalized defaults. All the `getattr` calls only apply a default if nothing higher up in the chain had set a value already.

Reply:
Good point!